contexts/TelnyxMeetContext.ts

```diff
@@ -38,6 +41,9 @@ const TelnyxMeetContext = React.createContext<{
   setVideoInputDeviceId: (
     value: React.SetStateAction<string | undefined>
   ) => {},
+  setParticipantJoined: (
+    value: React.SetStateAction<string | undefined>
+  ) => {},
```

Comment:
I would suggest not to use React.Context here. Especially because this is only storing the latest participant who joined and this could cause race conditions and you'll probably miss updates for certain participants.

Reply:
Instead of doing this, we can set the phone number of the participant we expect to join the room. This value can be read in the `participant_joined` event in the `room` hook. If it matches the participant's username then we can set the participant to have joined.

You can also use an object here, something like ...

```ts
invitedParticipants: {
  "+353xxxxxx": false, // false could mean that they haven't joined yet
  "+44xxxxxxx"" true, // true could mean that they have joined
}
```

In the `participant_joined` event you can call a function that is supplied by the context to update this object based on the username of the participant that joined.

Reply:
I'm glad you've pointed this out, @arghav. I wasn't sure myself.

Though we should discuss this further, because the events in the `room` hook are inside a `useEffect` that only runs once -- meaning it won't have access to any `invitedParticipants` updates later.

Reply:
As discussed, since the `invitedParticipants` object is only used by one component, it should be moved out of the context.

Now in [2265080](https://github.com/team-telnyx/telnyx-meet/pull/70/commits/226508096b6462b1d88efe0ec51b23cb83112ef3), a callback function to update this object on `participant_joined` is passed down to `room` hooks, and it is too passed down to the `InviteParticipant` component so we can keep track if the invited participant has joined the session.